trpcultivate_phenotypes/src/Form/PhenodataBackupForm.php

```diff
@@ -0,0 +1,327 @@
+
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('config.factory'),
+      $container->get('tripal.importer'),
+      $container->get('current_user'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(array $form, FormStateInterface $form_state): array {
+
+    $form = parent::form($form, $form_state);
+
+    $entity = $this->getEntity();
+
+    // Entity configuration ID number.
+    $form['id'] = [
+      '#type' => 'hidden',
+      '#default_value' => $entity->id(),
+      '#value' => uniqid(),
+    ];
+
+    if ($entity->isNew()) {
+      // Synchronize the file types defined by the importer that will be the
+      // main source of future backup data collection file.
+      $importer_plugin_definition = $this->tripal_importer
+        ->getDefinition('trpcultivate-phenotypes-share-importer');
+      $importer_file_extension = implode(' ', $importer_plugin_definition['file_types']);
+
+      $config_backup_dir = $this->service_ConfigFactory
+        ->get('trpcultivate_phenotypes.settings')
+        ->get('trpcultivate.phenotypes.directory.data_backup');
+      $backup_dir = $config_backup_dir ?? 'public:://phenotype-backups/';
+
+      $form['backup_file'] = [
+        '#type' => 'managed_file',
+        '#title' => 'Data File',
+        '#description' => $this->t('Select data file to backup. Only [@ext] file extensions are allowed.', ['@ext' => $importer_file_extension]),
+        '#upload_location' => $backup_dir,
+        '#multiple' => FALSE,
+        '#required' => TRUE,
+        '#upload_validators' => [
+          'FileExtension' => [
+            'extensions' => $importer_file_extension,
+          ],
+        ],
+      ];
+
+      // Restrict project autocomplete suggestions to project used as
+      // research_experiment Tripal content type.
+      $entity_type = $this->service_EntityTypeManager
+        ->getStorage('tripal_entity_type')
+        ->load('research_experiment');
+
+      $type_id = 0;
+      if ($entity_type) {
+        $entity_type_term_internalid = $entity_type->getTerm()
+          ->getInternalId();
+
+        // 0 value will suggest all projects.
+        $type_id = $entity_type_term_internalid;
+      }
+
+      $form['project_name'] = [
+        '#type' => 'textfield',
+        '#title' => 'Project/Experiment Name',
+        '#description' => $this->t('Select the project name the data file is specific to.'),
+        '#description_display' => 'after',
+        '#required' => TRUE,
+        '#attributes' => ['placeholder' => 'Project/Experiment Name'],
+        '#autocomplete_route_name' => 'tripal_chado.generic_autocomplete',
+        '#autocomplete_route_parameters' => [
+          'type_id' => $type_id,
+          'match_limit' => 5,
+          'base_table' => 'project',
+          'column_name' => 'name',
+          'type_column' => 'type_id',
+          'property_table' => 'project',
+        ],
+      ];
```

Comment:
Same as with the file.


```suggestion
      $project_id = (int) $entity->get('project_id');
      $project_name = ($project_id) ? ChadoProjectAutocompleteController::getProjectName($project_id): '';
      $form['project_name'] = [
        '#type' => 'textfield',
        '#title' => 'Project/Experiment Name',
        '#description' => $this->t('Select the project name the data file is specific to.'),
        '#description_display' => 'after',
        '#required' => TRUE,
        '#attributes' => ['placeholder' => 'Project/Experiment Name'],
        '#autocomplete_route_name' => 'tripal_chado.generic_autocomplete',
        '#autocomplete_route_parameters' => [
          'type_id' => $type_id,
          'match_limit' => 5,
          'base_table' => 'project',
          'column_name' => 'name',
          'type_column' => 'type_id',
          'property_table' => 'project',
        ],
        '#value' => ($project_id) ? "$project_name ($project_id)" : NULL,
        '#disabled' => ($project_id) ? TRUE : FALSE,
      ];
```

Reply:
implemented